### ISSUE TITLE
[RNAsubopt] redundant output for deltaE close to INF

### ISSUE THREAD

**Issue by martin-raden** [ViennaRNA/ViennaRNA#40]
I got some headache due to unexpected redundant output of RNAsubopt. In the end, it boiled down to the fact, that you are defining INF=100000 and my deltaEnergy was slightly above that threshold. Here the infamous example:

```[bash]
 --> RNA="GGGAAUUAUUGUUCCCUGAGAGCGGUAGUUCUC"
 --> echo $RNA |  RNAsubopt --deltaEnergy=100013.50 | wc -l
29763945
 --> echo $RNA | RNAsubopt --deltaEnergy=999 | wc -l
29759372
```

The latter is the correct number. The deltaE was computed as `99999 - MFE` where the mfe is -14.50 for this RNA.

Thus, your input sanity checks in [`ViennaRNA/subopt.c`](https://github.com/ViennaRNA/ViennaRNA/blob/34d23a07f633e8fe2366877e1e7e1d9a232dcd96/src/ViennaRNA/subopt.c#L857) are not triggered but it seems a threshold value around INF causes some trouble in the subopt routines...

While this is an artifact of my extremly high deltaE (to enumerate the whole structure space) and can be countered by using eg 9999 instead of 99999 for the deltaE computation, I still thought it important enough to report.. :smile: 

Best,
Martin

**Comment by RaumZeit** [comment]
thx! While the forward recursions to fill the DP matrices should refrain from composing any structures where parts are evaluated as `INF`, the suboptimal structure backtracking functions do not all adhere to this convention. I'll see that (a) this will be changed in the future, and (b) that `deltaE` values close to or larger than `INF` will not be consiered

**Comment by RaumZeit** [comment]
Fixed with version `2.4.10`
